### DISCUSSION TITLE
hibernate-reactive + vertx + aws iam

### DISCUSSION THREAD

**Discussion by vozdelib** [aws/aws-advanced-jdbc-wrapper#750] [Q&A]
Hello,
Does anyone know if there is a way to use IAM authentication with Vertx, please?
I'm trying to change the way my service authenticates with our RDS Mysql, I'm probably missing some configuration or it is maybe not possible? It works with a normal user + password, but I can't find a way to use passwordless IAM authentication.
Thank you.

**Comment by brunos-bq** [comment]
Hello @vozdelib,

Thank you for reaching out.

In the `examples` folder of our repository, you will find a [sample application](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/f3a0900761336ea8d5d4524bee7bd96c64d521d8/examples/VertxExample/src/main/java/com/example/starter/MainVerticle.java) that instructs on how to use the AWS JDBC Driver with vertx.

In that example, if one would want to use IAM authentication, they would have to:

- Enable the IAM Authentication plugin in the Connection String URI from `jdbc:aws-wrapper:postgresql://db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com:5432/postgres` to `jdbc:aws-wrapper:postgresql://db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com:5432/postgres?wrapperPlugins=iam,[otherPlugins],failover,efm`

- Then, in the same configuration, update the username that will connect to the database to an IAM database user.

You will find more details on how to use IAM within the AWS JDBC Driver in our [documentation](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/using-plugins/UsingTheIamAuthenticationPlugin.md).

Please keep in mind that in order to use IAM Authentication with the AWS JDBC Driver, you will be required to have the [AWS Java SDK RDS v2.x](https://central.sonatype.com/artifact/software.amazon.awssdk/rds) separately included in the classpath of your application.

Thanks!

**Reply by vozdelib** [reply]
Thank you for taking your time to answer, 
I have already tried this, but this doesn't work. Forgive me, I'm quite a newbie with vertx and java world in general, so maybe i'm doing wrong something very basic.
But from my perspective it seems that hibernate-reactive uses https://vertx.io/docs/vertx-mysql-client/java/ and this seems very proprietary, without option to extend with plugins. It also doesn't seem to be using standard java JDBC Driver. So maybe it is the reason why this aws-advanced-jdbc-wrapper doesn't work. I don't know. Seems to me that I would have to create my own iam token generator and somehow plug it into the code. For my usecase this would be huge overkill. So unless I figure out some simple solution (like you proposed) for now sticking with normal usr+pwd authentication...

**Comment by brunos-bq** [comment]
Hello,

I am not sure i fully understand your last message: Are you able to use the AWS JDBC Driver with Vertx, but the IAM Authentication does not work as expected, or are you not able at all to use the AWS JDBC Driver?

**Reply by vozdelib** [reply]
I'm not able to use AWS JDBC Driver. Hibernate-reactive (https://github.com/hibernate/hibernate-reactive) uses vertx-mysql-client which is probably build specifically for this purpose to be reactive and is doing the network calls on its own, from what i can tell.

**Reply by davecramer** [reply]
Yes, you are correct. The JDBC driver is not reactive and is not compatible with this application.

**Comment by brunos-bq** [comment]
Hello @vozdelib,

Is there a way you could show us how your application instantiates the MySQL client and which configuration it is using? Perhaps with a simplified sample of your application that describes your actual workflow, we would be able to better understand how to use the AWS JDBC Driver in that situation.
